dji_asdk_to_python/precision_landing/ISOChanger.py

```diff
@@ -0,0 +1,157 @@
+        return ISO_CONFIGS[new_index]
+
+    def _has_iso_changed(self, iso_config: ISOConfig) -> bool:
+        """
+        Checks wheter the drone has changed its ISO configuration to
+        the one requested in 'iso_config'.
+
+        Args:
+            iso_config (ndarray): the ISO configuration that drone should
+            have.
+
+        Returns:
+            bool: True in case the drone changed its ISO configuration
+            to 'iso_config' ones, False otherwise.
+        """
+
+        cond: bool = self._camera.getISO() == iso_config.iso
+        cond = cond and (self._camera.getShutterSpeed() == iso_config.shutter_speed)
+        cond = cond and (self._camera.getExposureMode() == iso_config.exposure_mode)
+        return cond
+
+    def is_current_iso_suitable(self, frame: ndarray) -> bool:
+        """
+        Check if the current ISO is suitable by comparing if
+        the brigthness rate of a current frame is inn the range
+        [LOWER_BRIGHTNESS_RATE_BOUND, UPPER_BRIGHTNESS_RATE_BOUND]
+
+        Args:
+            frame (ndarray): latest frame obtained.
+
+        Returns:
+            bool: True in case the changed ISO is in the range
+            [LOWER_BRIGHTNESS_RATE_BOUND, UPPER_BRIGHTNESS_RATE_BOUND],
+            False otherwise
+        """
+
+        brightness_rate: float = calculate_image_brightness(frame)
+        cond: bool = brightness_rate < UPPER_BRIGHTNESS_RATE_BOUND
+        cond = cond and brightness_rate > LOWER_BRIGHTNESS_RATE_BOUND
+        return cond
+
+    def _change_iso(self, frame: ndarray) -> None:
+        """
+        Changes ISO to a more suitable one and waits until the drone has
+        applied the changes.
+
+        Args:
+            frame (ndarray): latest frame obtained.
+        """
+
+        next_iso: ISOConfig = self._find_next_likely_suitable_iso(frame)
+
+        CameraSetting(
+            camera=self._camera,
+            display_mode=next_iso.display_mode,
+            exposure_mode=next_iso.exposure_mode,
+            iso=next_iso.iso,
+            shutter_speed=next_iso.shutter_speed
+        ).set_camera_setting()
+
+        while not self._has_iso_changed(next_iso):
+            time.sleep(0.1)
+
+        time.sleep(1)  # wait until changes has been applied
+        self._is_iso_changing = False
+
+    def change_iso(self, frame: ndarray, force_change: bool = False) -> None:
+        """
+        Changes the ISO until it has found a suitable configuration to be
+        able to detect arucos.
+
+        Args:
+            frame (ndarray): latest frame obtained.
+            force_change (bool): tells the function it has to change the ISO
+            regardless of whether it's suitable or not.
+        """
+        self._is_iso_changing = True
+        has_suitable_iso: bool = self.is_current_iso_suitable(frame)
+        if has_suitable_iso and force_change:
+            has_suitable_iso = False
+
+        if has_suitable_iso:
+            return
```

Comment:
No deberia tener un return si se supone que no retorna nada. Se puede mejorar los condicionales.
```suggestion
    self._is_iso_changing = True

    if not (self.is_current_iso_suitable(frame) and force_change):
        self._change_iso(frame)
```

Reply:
Esta sugerencia es para la funcion completa

Reply:
El return solo rompe la funcion. Es una funcion vacia que cambia el ISO bajo ciertas condiciones. Segun como esta escrita, veo que hay mayor complejidad temporal ya que estas evaluando mas instrucciones... Esta funcion puede salir en menos lineas de codigo, mas legible, y sin la necesidad de incluir un return porque igual si no entra en el if, no ocurrira la siguiente linea.

Reply:
El condicional no sirve, porque force change no siempre es True, a veces el cambio no debe ser forzado y en ese caso, la función no cumpliría con lo cometido.